src/CMakeLists.txt

```diff
@@ -459,6 +459,20 @@ ELSE()
     $<INSTALL_INTERFACE:${KOKKOSKERNELS_HEADER_INSTALL_DIR}>)
 ENDIF()
 
+IF (KOKKOS_ENABLE_SYCL)
+  SET(KOKKOSKERNELS_INTEL_ARCHS ${Kokkos_ARCH})
+  LIST(FILTER KOKKOSKERNELS_INTEL_ARCHS INCLUDE REGEX ".*INTEL.*")
+  LIST(LENGTH KOKKOSKERNELS_INTEL_ARCHS KOKKOSKERNELS_INTEL_ARCHS_NUM)
+  IF(KOKKOSKERNELS_INTEL_ARCHS_NUM GREATER_EQUAL 1)
+    IF (NOT BUILD_SHARED_LIBS)
+      MESSAGE(SEND_ERROR
+        "At the moment, we require KokkosKernels (and Kokkos) to be built as "
+        "shared libraries to allow querying free and total device memory!"
+      )
+    ENDIF()
```

Comment:
Note that this is a new restriction that may surprise users already using `SYCL`.